libbeat/beat/event.go

```diff
@@ -54,6 +54,19 @@ func (e *Event) SetID(id string) {
 	e.Meta["_id"] = id
 }
 
+func (e *Event) GetMetaStringValue(key string) (string, error) {
```

Comment:
Nit: The function can be simplified to this:

```golang
tmp, err := e.GetValue("@metadata." + key)
if err == nil {
    if s, ok := tmp.(string); ok {
	return s, nil
    }
}
return "", nil
```

Reply:
how about (requires no alloc):

```
tmp, err := e.Meta.GetValue(key)
if err != nil {
  return "", err
}

if s, ok := tmp.(string); ok {
  return s, nil
}

return "", nil
```

All in all I'd prefer to keep the interface of `Event` as small as possible and provide helpers as functions instead of methods.

Reply:
This is even better. \o/

Reply:
Thanks for the feedback. I've used the version from @urso . Nice to see this level of attention to detail!